### PR TITLE
Update lstoll/oidc

### PR DESCRIPTION
**Pull request by lstoll**
Latest version, for JWT access tokens and PKCE. Plus, it has a bundled static client manager - just use that, and drop the local one.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,11 +11,12 @@ require (
 	github.com/google/uuid v1.6.0
 	github.com/justinas/nosurf v1.1.1
 	github.com/lstoll/cookiesession v0.0.0-20240302214249-23c5d01c3fb9
-	github.com/lstoll/oidc v0.0.0-20240302161652-ec893a64705d
+	github.com/lstoll/oidc v1.0.0-alpha.1
```
Comment (sr) [inline go.mod:14]: v1 when

File: config.go [hunk 2]
```diff
@@ -48,16 +43,10 @@ func loadConfig(file []byte, cfg *config) error {
...
+		for ii, c := range cfg.Issuer[i].Clients {
+			if c.ID == "" {
 				return fmt.Errorf("issuer %s client %d must set clientID", parsed, ii)
 			}
-			if len(c.ClientSecret) == 0 {
-				return fmt.Errorf("issuer %s client %d must set at least one clientSecrets", parsed, ii)
-			}
-			if len(c.RedirectURL) == 0 {
-				return fmt.Errorf("issuer %s client %d must set at least one redirectURLs", parsed, ii)
-			}
```
Comment (sr) [inline config.go:60]: Would be nice to validate early still; could this call IsValidClientID etc at the end?

**Review by sr — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (lstoll) [reply go.mod:14]: when I stop fucking around lol

File: config.go (hunk 2 above)
Comment (lstoll) [reply config.go:60]: redirect URL isn't always needed - I'll add some better validation to the upstream library.

**Pull request merged**